### ISSUE TITLE
Next sentence prediction configuration

### ISSUE THREAD

**Issue by Alexadar** [jina-ai/clip-as-service#285]
Hi!
Im trying to use this package for next sentence prediction from pre-trained model

bert-serving-start -model_dir=/models/uncased_L-24_H-1024_A-16/ -num_worker=1 -max_seq_len=512 -max_batch_size=128 -gpu_memory_fraction=1

0. Which pooling strategy should i use for next sentence prediction?

1. Server is extremelly slow - 2 sentences per sec. On same server by using 
pytorch_pretrained_bert i have ~150 sentence pair/sec . What to do? Is it a bug?

2. I have returned a vector of floats to client. How to calculate label next / notNext from that vector?

Thanks!

**Comment by hanxiao** [comment]
1. I'm afraid your GPU is not utilized. So please read the server log, check the GPU usage of `nvidia-smi`.
2. Set `-max_seq_len=NONE` instead of 512

**Comment by Alexadar** [comment]
1.  nvidia-smi -l 1 Showed its utilized 99% GPU and 100% RAM
2. Thanx!
3. How to calculate nextSentence label from output?

**Comment by bxclib2** [comment]
Hi, I experienced the same issue.  The symptoms are exactly the same as he described. Pytorch_pretrained_bert is about 100 times faster in my machine than bert as service.  and I am totally sure my GPU is working hard just by the sound of the fan.....

My text is Kaggle Gender pronoun competition,  and it costs 1.5 hour to deal the train data in bert as service, but in pytorch_pretrained_bert it takes only 15s.

I am really curious what happened in bert as service...

**Comment by pbakhtin** [comment]
Bert-as-service is great tool and I also want to use it for sentence prediction. Any tips on how to transform CLS embedding (that we get for a pair of sentences) to probability of sentence "b" being continuation of sentence "a"?

I got as deep as this code, but I don't understand what it does:
_, pooled_output = model.bert(tokens_tensor,segments_tensors, attention_mask=None,
                                 output_all_encoded_layers=False)
seq_relationship_score = model.cls(pooled_output)

**Comment by Kesin1** [comment]
@Alexadar Hey, I used the BertPooler and OnlyNSPHead from the pretrained Bert model from pytorch_transformers and put it after the bert-as-a-server output with -pooling-strategy None and -pooling_layer -1 ... that gave me the same results as with the pytorch_transformer class BertForNextSentencePrediction
